src/main/kotlin/org/ionproject/integration/utils/Try.kt

```diff
@@ -32,26 +37,29 @@ sealed class Try<out T> {
         override fun <R> match(success: (T) -> R, error: (Exception) -> R) = success(value)
     }
 
-    class Error(val e: Exception) : Try<Nothing>() {
+    class Error<out T>(val e: Exception) : Try<T>() {
```

Comment:
Couldn't T be bound to be an exception, like so: class Error <out T: Exception> (val e : Exception) ?

Reply:
No because then R would also need to be an exception (because it's the return of map, flatmap,..) and that would break Value